### PR TITLE
[UX] Modification by filter

### PR DESCRIPTION
**Pull request by thangqp**
Two UX aspects enhanced in this PR:

1. Auto-hide button Delete while the cursor is not hover on a row => impacts on all creations/modifications which support to edit properties and some others particulars infos. 
2. Set the dialogue height fixed 95% view port

### CODE REVIEW

**Review by Mathieu-Deharbe — APPROVED** [review]
OK for me.

I don't know if that part was supposed to change as well but just in case, diagrams trash lines are always visible, hovered or not : 

poubelles

File: src/components/utils/rhf-inputs/expandable-input/deletable-row.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ */
+import { PropsWithChildren, useState } from 'react';
+import { useIntl } from 'react-intl';
+import { Grid, IconButton, Tooltip } from '@mui/material';
+import RestoreFromTrashIcon from '@mui/icons-material/RestoreFromTrash';
+import DeleteIcon from '@mui/icons-material/Delete';
+
+export interface DeletableRowProps extends PropsWithChildren {
+    alignItems: string;
+    onDeleteClick: () => void;
```
[the remark this answers is not in the document]
Comment (thangqp) [reply src/components/utils/rhf-inputs/expandable-input/deletable-row.tsx:15]: YES!

**Comment by thangqp** [comment]
> OK for me.
> 
> I don't know if that part was supposed to change as well but just in case, diagrams trash lines are always visible, hovered or not :
> 
> poubelles

It does not use the same component ExpandableInput => out of scope of this PR

File: src/components/utils/rhf-inputs/expandable-input/deletable-row.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (thangqp) [reply src/components/utils/rhf-inputs/expandable-input/deletable-row.tsx]: DONE in https://github.com/gridsuite/gridstudy-app/pull/2343/commits/7c1079193be40395f2934dfb6142cc033d71426b

File: src/components/utils/rhf-inputs/expandable-input/index.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+export * from './expandable-input';
```
Comment (ghazwarhili) [inline src/components/utils/rhf-inputs/expandable-input/index.ts:8]: You can import it into the index.tsx file of src/component
why you need to create a new file to import this

Comment (thangqp) [reply src/components/utils/rhf-inputs/expandable-input/index.ts:8]: This is the new convention of importing as commons-ui, we will do the same for other components

File: src/components/utils/rhf-inputs/expandable-input/expandable-input.tsx [hunk 3]
```diff
@@ -64,25 +60,22 @@ const ExpandableInput: FunctionComponent<ExpandableInputProps> = ({
...
+                        alignItems={alignItems}
+                        onClick={() => {
+                            if (deleteCallback) {
+                                if (deleteCallback(idx) === true) {
+                                    remove(idx);
+                                }
+                            } else {
+                                remove(idx);
+                            }
+                        }}
```
[the remark this answers is not in the document]
Comment (ghazwarhili) [reply src/components/utils/rhf-inputs/expandable-input/expandable-input.tsx:74]: could you explain this if() {} else {} to do the same thing remove (idx)

File: src/components/utils/rhf-inputs/expandable-input/deletable-row.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+            container
+            spacing={2}
+            item
+            alignItems={alignItems}
+            onMouseEnter={() => setIsHover(true)}
+            onMouseLeave={() => setIsHover(false)}
+        >
+            {children}
+            <Grid item xs={1}>
+                {isHover && (
```
Comment (ghazwarhili) [inline src/components/utils/rhf-inputs/expandable-input/deletable-row.tsx:34]: isMouseHover ?

Comment (thangqp) [reply src/components/utils/rhf-inputs/expandable-input/deletable-row.tsx]: DONE in https://github.com/gridsuite/gridstudy-app/pull/2343/commits/9a773e46161bfa54e3ce3e267203bb1f5d6636ec

**Review by ghazwarhili — APPROVED** [review]

**Pull request merged**